File: sprint4/By lessons/lesson10code2/lesson10code2.py
def effectiveSolution(A, x):
    history = set()
    n = len(x)
    xSorted = sorted(x)
    triples = set()
    for i in range(n):
        for j in range(i + 1, n):
            target = A - xSorted[i] - xSorted[j]
            if target in history:
                triples.add((target, xSorted[i], xSorted[j]))
            history.add(xSorted[i])
    return triples

File: sprint4/By lessons/lesson10code2/lesson10code2.swift
struct Triple: Hashable {
    let first: Int
    let second: Int
    let third: Int
}

func effectiveSolution(_ A: Int, _ x: [Int]) -> Set<Triple> {
    var history: Set<Int> = []
    let n = x.count
    let xSorted = x.sorted()
    var triples: Set<Triple> = []

    for i in 0..<n {
        for j in (i + 1)..<n {
            let target = A - xSorted[i] - xSorted[j]
            if history.contains(target) {
                let triple = Triple(first: target, second: xSorted[i], third: xSorted[j])
                triples.insert(triple)
            }
            history.insert(xSorted[i])
        }
    }

    return triples
}
